trident/firefly_generator.py

```diff
@@ -0,0 +1,361 @@
+"""
+FireflyGenerator class and member functions.
+
+"""
+
+import numpy as np
+import os
+
+from yt.loaders import \
+    load
+from yt.data_objects.data_containers import \
+    YTDataContainer
+from yt.data_objects.static_output import \
+    Dataset
+from yt.funcs import \
+    mylog, \
+    YTArray
+from yt.utilities.exceptions import \
+    YTFieldNotFound
+from yt.utilities.on_demand_imports import \
+    _firefly
+
+from trident.config import \
+    ion_table_dir, \
+    ion_table_file, \
+    ion_table_filepath
+from trident.ion_balance import \
+    add_ion_number_density_field, \
+    atomic_mass
+from trident.line_database import \
+    LineDatabase
+
+class FireflyGenerator( object ):
+    """
+    ## OLD TEXT COMMENTED OUT ##
```

Comment:
Remove commented out docstrings for SpectrumGenerator and replace with description of FireFlyGenerator functionality and arguments.